### PR TITLE
Feature/dbi and refactor

### PR DESCRIPTION
**Pull request by nbc**
Hi Damien,

Last PR for the moment : 

The main work is around refactoring table_to_parquet and then all other *_to_parquet to extract the reading writing logic into two functions : 

- write_parquet_at_once if input is in one datafile/tibble
- write_parquet_by_chunk if input must be read/write by chunk.

write_parquet_at_once is already used everywhere
write_parquet_by_chunk should be usable in almost all *_to_parquet functions (but dbi_to_parquet)

path_to_parquet argument should be a file path and not a directory. For the moment I choose to issue a warning and create a file path with file name as paste0(basename(path_to_parquet), ".parquet") but we could choose to fail.

I'm not sure that's ready for delivery : there's some rough edges in documentation and I need to let the dust settle a bit to think clearly.

I let you read/play with this.

Some cleaning : 

- a bit of work on expect_parquet : I add a test on file's number
- remove the label removing
- I add two skip_if_offline because of the CRAN warning

Regards

### CODE REVIEW

File: R/table_to_parquet.R [hunk 1]
```diff
@@ -104,45 +104,45 @@
...
 #' # and conversion to multiple files with zstd, compression level 10
 #'
-#' table_to_parquet(
-#'   path_to_table = system.file("examples","iris.sas7bdat", package = "haven"),
-#'   path_to_parquet = tempdir(),
-#'   max_rows = 50,
-#'   compression = "zstd",
-#'   compression_level = 10
-#' )
+#' if (isTRUE(arrow::arrow_info()$capabilities[['zstd']])) {
```
Comment (ddotta) [inline R/table_to_parquet.R:123]: ok it's a very smart way to avoid the `\dontrun{}`

Comment (nbc) [reply R/table_to_parquet.R:123]: Yes indeed, but it's not really my idea :) I took the idea in https://r-pkgs.org/man.html#sec-man-examples-dependencies-conditional-execution.

Comment (ddotta) [reply R/table_to_parquet.R:123]: Waou cool !

**Comment by ddotta** [comment]
@nbc I've been testing and playing with the PR a bit as you say :)

I agree to leave as you suggest for the path_to_parquet behaviour: create a file with the same name as the initial file and leave a small warning for users.

Concerning the CRAN warning and as we may be on holiday in the next few days, I would like to see this PR merged before 30/04 (for the part that uses the skip_if_offline).

Do you still want to take some time to refine the documentation?

**Comment by nbc** [comment]
Hi @ddotta ,

> Do you still want to take some time to refine the documentation?

If the use documentation is ok for you you can merge. I can do some cleaning on developer's documentation later when I have more time. It's no like we were 30 to develop :)

And "bonnes vacances" !

**Comment by ddotta** [comment]
Ok @nbc we can take care of further documentation later.

> And "bonnes vacances" !

Merci :)

**Pull request merged**